lib/foreman_tasks/tasks/test.rake

```diff
@@ -1,14 +1,16 @@
 namespace :test do
-  task :foreman_tasks => 'db:test:prepare' do
-    test_task = Rake::TestTask.new('foreman_tasks_test_task') do |t|
-      t.libs << ['test', "#{ForemanTasks::Engine.root}/test"]
-      t.test_files = Dir["#{ForemanTasks::Engine.root}/test/**/*_test.rb"]
-      t.verbose = true
-      t.warning = false
-    end
-
-    Rake::Task[test_task.name].invoke
+  desc 'Foreman Tasks plugin tests'
+  Rake::TestTask.new(foreman_tasks: 'db:test:prepare') do |t|
```

Comment:
One downside of this is that it'll also be included in `foreman-rake` and executed, which slows things down. The previous approach only defined a very simple task and loaded things on demand.

We may want to exclude this file from the gem or RPMs/debs.